tests/parser/functions/test_convert.py

```diff
@@ -466,22 +465,6 @@ def test_convert() -> {o_typ}:
 
     c1_exception = None
     skip_c1 = False
-    if i_typ in INTEGER_TYPES and o_typ in INTEGER_TYPES - {"uint256"}:
-        # Skip conversion of positive integer literals because compiler reads them
-        # as target type.
-        if val >= 0:
-            c1_exception = InvalidType
-
-    if i_typ in SIGNED_INTEGER_TYPES and o_typ in SIGNED_INTEGER_TYPES and val < 0:
-        # similar, skip conversion of negative integer literals because compiler
-        # infers them as target type.
-        c1_exception = InvalidType
-
-    if i_typ.startswith(("int", "uint")) and o_typ.startswith("bytes"):
-        # integer literals get upcasted to uint256 / int256 types, so the convert
-        # will not compile unless it is bytes32
-        if o_typ != "bytes32":
-            c1_exception = TypeMismatch
```

Comment:
Are these cases now handled by the compiler better?

Reply:
Yes, these are now valid conversions (or typecasting) with the introduction of type inference, so exceptions are no longer thrown.